### PR TITLE
Data explorer editor redux

### PR DESCRIPTION
**Pull request by j-f1**
…and a bunch of other things. [The whitespace-insensitive diff](https://github.com/Charcoal-SE/metasmoke/pull/212/files?w=1) might be cleaner.

### CODE REVIEW

File: app/javascript/data.js [hunk 1]
```diff
@@ -140,23 +149,56 @@ const validateDataset = () => {
...
+  editor.$blockScrolling = Infinity;
   editor.getSession().setMode('ace/mode/javascript');
   editor.setOptions({
     minLines: 15,
     maxLines: 30,
     useSoftTabs: true,
-    tabSize: 4,
-    printMarginColumn: 120
+    tabSize: 2,
+    printMarginColumn: 80
```
Comment (ArtOfCode-) [inline app/javascript/data.js]: What was wrong with 4/120?

**Comment by ArtOfCode-** [comment]
Why is using `eval` here a problem? As far as I can tell, there's no security risk here because the user code never goes anywhere but the author's client.

File: app/javascript/data.js (hunk 1 above)
Comment (j-f1) [inline app/javascript/data.js]: The JS code in the app uses 2 spaces. I’ll change the `printMarginColumn` part back.

**Comment by j-f1** [comment]
Fixed @ArtOfCode-

**Pull request merged**